templates-blocks/grid-links/grid-links.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * UDS Grid Links
+ *
+ * @package UDS WordPress Theme
+ */
+
+$source = get_field( 'uds_grid_links_source' );
+$columns = get_field( 'uds_grid_links_columns' );
+$breakpoint = get_field( 'uds_grid_links_breakpoint' );
+$color = get_field( 'uds_grid_links_color' );
+
+
+// Retrieve additional classes from the 'advanced' field in the editor.
+$additional_classes = '';
+if ( ! empty( $block['className'] ) ) {
+	$additional_classes = $block['className'];
+}
+
+// Add the required text-color modifier to the additional classes string.
+if ( 'none' !== $color ) {
+	$additional_classes = $color . ' ' . $additional_classes;
+}
+
+// Add the required column modifier to the additional classes string.
+if ( 'mobile' !== $columns ) {
+	$additional_classes = $columns . ' ' . $additional_classes;
+}
+
+echo '<div class="uds-grid-links ' . $additional_classes . '">';
+
+// If then structure determines the actual <a> tags produced. Indicated with user choice for the source of the links.
+// This field is a radio button with a default setting, so no need to see if it exists.
+if ( 'arbitrary' === $source ) {
+
+	// Check the rows of the repeater, build the links.
+	if ( have_rows( 'uds_grid_links_created' ) ) :
+
+		while ( have_rows( 'uds_grid_links_created' ) ) :
+			the_row();
+
+			$icon = get_sub_field( 'uds_grid_links_created_icon' );
+			$gridlink = get_sub_field( 'uds_grid_links_created_link' );
+			$linkstring = '';
+
+			if ( ! empty( $icon ) ) {
+				$linkstring .= '<span class="fa-fw ' . $icon . '"></span>';
+			}
+
+			$link_url = $gridlink['url'];
+			$link_title = $gridlink['title'];
+			$link_target = $gridlink['target'] ? $gridlink['target'] : '_self';
+
```

Comment:
I noticed that when I'm on the preview mode, and trying to add a new row from the editor box on the right side, if the top row had an icon, and I try to add a new row, the block breaks, and it displays  the error: 
![Screen Shot 2021-06-10 at 3 16 13 PM](https://user-images.githubusercontent.com/15880606/121605253-effbcf00-ca00-11eb-96a7-f4fb7da8877b.png)

Then in the page preview I see the following warnings for all 3 lines (50, 51 and 52):
Warning: Illegal string offset ‘url’ in /app/wp-content/themes/UDS-WordPress-Theme/templates-blocks/grid-links/grid-links.php on line 50 
Notice: Uninitialized string offset: 0 in /app/wp-content/themes/UDS-WordPress-Theme/templates-blocks/grid-links/grid-links.php on line 50

Reply:
This took me a bit to figure out. Within the `have_rows` loop in the PHP output, the variables for storing and building each grid link were not being reset with each iteration. As a result, there were intermittent errors that only occurred after the build of the initial link.